### DISCUSSION TITLE
Improving styling options

### DISCUSSION THREAD

**Discussion by megagrump** [flamendless/Slab#111] [Ideas]
So my idea to make styling more customizable is through use of [9-sliced](https://en.wikipedia.org/wiki/9-slice_scaling) images.

Most of the widgets currently use simple rectangles in `line` and `fill` mode. If we replaced those with 9-patches, making a new theme would become as simple as drawing an image with the new look, and defining a bunch of rectangles.

If you want to get fancy you can even use a Canvas as the slice source and get animated styles "for free" (as in, no special support is required in the library).

This is, in my opinion, an elegant solution: it makes creating new styles very simple, does not require writing or changing code to create a theme, and it's relatively easy to implement.

It does require a good caching strategy though, because you don't want to recreate a bunch of SpriteBatches in every frame. That's probably the hardest part of the implementation.

**Comment by megagrump** [comment]
Yo @flamendless did you see this? Any thoughts?

**Comment by flamendless** [comment]
Oh problem with the mobile version it seems. Now i see it.

Hmmm 9-slice is a good idea for theming. I am all for it.
